dali/python/nvidia/dali/pipeline.py

```diff
@@ -177,6 +183,28 @@ def epoch_size(self, name = None):
             return self._pipe.reader_meta(name)["epoch_size_padded"]
         return {name : v["epoch_size_padded"] for k, v in self._pipe.reader_meta()}
 
+    def executor_statistics(self):
+        """Returns provided pipeline executor statistics metadata as a dictionary.
+        Each key in the dictionary is the operator name. To enable it use ``executor_statistics``
+
+        Available metadata keys for each operator:
+
+        ``real_memory_size``:     list of memory sizes that is used by each output of the operator;
+                                  index in the list corresponds to the output index
+
+        ``max_real_memory_size``: list of maximum tensor size that is used by each output of the operator;
+                                  index in the list corresponds to the output index
+
```

Comment:
Only small nitpick: this is max for non-contiguous buffers and average for contiguous. In the case of real_memory we theoretically could calculate the volume of the samples but I don't think it's worth.